### ISSUE TITLE
ECS Task Definition: Updating network mode should force replacement

### ISSUE THREAD

**Issue by ryanpodonnell1** [pulumi/pulumi-aws-native#733]
### What happened?

```
pulumi version: v3.45.0
aws-native plugin: 0.43.0
sdk: "github.com/pulumi/pulumi-aws-native/sdk/go/aws/ecs"
```

ECS Task definition was deployed without a network mode specified.

Updated ECS Task definition network mode to `awsvpc` and resulted in the following error when pulumi attempted to update the task definition:

preview:

```
    ~ aws-native:ecs:TaskDefinition: (update)
        [id=arn:aws:ecs:us-west-2:<redacted>:task-definition/LJWk46EDDqfqcJOMut5ZQB2iU-aWZI8HIi0OLT:1]
        [urn=urn:pulumi:cloudsec-vulnerability-management-infra.vuln-management.rpo::findings-generator::aws-native:ecs:TaskDefinition::vuln-mgmt-infra-relay]
      + networkMode: "awsvpc"
```

error
```sh
  aws-native:ecs:TaskDefinition (vuln-mgmt-infra-relay):
    error: operation error CloudControl: UpdateResource, https response error StatusCode: 400, RequestID: 476719c2-544a-4e79-acfa-a66dfad4cde6, NotUpdatableException: Invalid patch update: createOnlyProperties [/properties/NetworkMode] cannot be updated
```

### Steps to reproduce

* create a task definition with a service without a network mode specified
* update network mode to `awsvpc`

### Expected Behavior

* recreate the task definition with the AWSVPC mode 

### Actual Behavior

* pulumi attempted to update the task definition rather than replace it

### Output of `pulumi about`

```
CLI          
Version      3.45.0
Go Version   go1.19.2
Go Compiler  gc

Plugins
NAME        VERSION
aws         5.21.1
aws         4.38.1
aws-native  0.43.0
go          unknown
```

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by ryanpodonnell1** [comment]
just ran into another issue when attempting to update task definition with a new ECR image:

```
Diagnostics:
  pulumi:pulumi:Stack (findings-generator-cloudsec-vulnerability-management-infra.vuln-management.rpo):
    error: update failed
 
  aws-native:ecs:TaskDefinition (vuln-mgmt-infra-relay):
    error: operation error CloudControl: UpdateResource, https response error StatusCode: 400, RequestID: 684216de-b92d-43f1-b4cb-708662e91312, NotUpdatableException: Invalid patch update: createOnlyProperties [/properties/ContainerDefinitions] cannot be updated
    ```

**Comment by ryanpodonnell1** [comment]
relevant code:

https://github.com/hashicorp/terraform-provider-aws/blob/08e008d7d40a5c8aabf8d30e462e6a3e779a6829/internal/service/ecs/task_definition.go#L62

**Comment by roothorp** [comment]
Hi @ryanpodonnell1, thanks for the issue! This looks to be a problem with how we're assessing what can and can't be updated in place (and what requires the resource to be replaced). I've raised the above issue (#736) to track this getting resolved.

**Comment by mjeffryes** [comment]
This should be resolved by #1053; changes to network mode will result in a replace rather than an update.
